### ISSUE TITLE
Feature request: C89 compatible version

### ISSUE THREAD

**Issue by wagnerma** [clwi/CWPack#7]
It would be quite usefull to have your library in C89 compatible code. For example the struct timespec is C11 and in our environment C11 is not allowed. And your macros (e.g. tryMove4) produces a lot of errors in C89 with "declaration may not appear after executable statement in block".

**Comment by clwi** [comment]
I made a branch "C89". You can check  if that works for you.

**Comment by smw-wagnerma** [comment]
Thanks, I did the changes already myself, but I tried your branch, it works fine.

One some small thing remains, I got some strange compiler warnings in the cw_unpack_next function with a Keil ARM compiler:

some warning:  #111-D: statement is unreachable

In the following places:
  case 0xd6:  getDDItemFix(4);
  case 0xd7:  getDDItemFix(8);
and
  at the end of the function at the
    return;

Ok, the last one, the return statement, seems useless, but the both others, I can't understand.

**Comment by clwi** [comment]
I think the warnings in 
case 0xd6: getDDItemFix(4);
case 0xd7: getDDItemFix(8);
comes from compiler optimisation of constant if expression. Inserted else statements to aid compiler. Pushed to C89 branch.

**Comment by smw-wagnerma** [comment]
The unreachable statement is fixed, thanks! Now I got "enumerated type mixed with another type", the first three warnings, I always had, the last ones are new.

```
base\mpack\cwpack.c(526): warning:  #188-D: enumerated type mixed with another type
                      unpack_context->item.type = *(int8_t*)p;
base\mpack\cwpack.c(544): warning:  #188-D: enumerated type mixed with another type
                      unpack_context->item.type = *(int8_t*)p;
base\mpack\cwpack.c(548): warning:  #188-D: enumerated type mixed with another type
                      unpack_context->item.type = *(int8_t*)p;
base\mpack\cwpack.c(575): warning:  #188-D: enumerated type mixed with another type
          case 0xd4:  getDDItemFix(1);                                                /* fixext 1 */
base\mpack\cwpack.c(576): warning:  #188-D: enumerated type mixed with another type
          case 0xd5:  getDDItemFix(2);                                                /* fixext 2 */
base\mpack\cwpack.c(577): warning:  #188-D: enumerated type mixed with another type
          case 0xd6:  getDDItemFix(4);                                                /* fixext 4 */
base\mpack\cwpack.c(578): warning:  #188-D: enumerated type mixed with another type
          case 0xd7:  getDDItemFix(8);                                                /* fixext 8 */
base\mpack\cwpack.c(579): warning:  #188-D: enumerated type mixed with another type
          case 0xd8:  getDDItemFix(16);                                               /* fixext 16 */

```
The type is an enum, if I add a cast, the compiler is happy, but I am not sure, if this is fine (I am not really a C expert):

                    unpack_context->item.type = (cwpack_item_types)*(int8_t*)p;

**Comment by clwi** [comment]
Solved with #13
